### DISCUSSION TITLE
Facebook RTMP

### DISCUSSION THREAD

**Discussion by yjjandyxy** [HaishinKit/HaishinKit.swift#1089] [Q&A]
I use the SDK with Facebook RTMP.  The first connection success but one second after Connect Closed, I can not push with Facebook RTMP.
The Facebook RTMP is  rtmps://live-api-s.facebook.com:443/rtmp/135738542601194?s_asc=1&s_bl=1&s_oil=2&s_psm=1&s_pub=1&s_sw=0&s_tids=1&s_vt=api-s&a=AbwDopRUK7vfeJSz

**Comment by shogo4405** [comment]
Related issue. https://github.com/shogo4405/HaishinKit.swift/issues/718, https://github.com/shogo4405/HaishinKit.swift/issues/877.

Please try this one.
```swift
rtmpConneciton.connect('rtmps://live-api-s.facebook.com:443/rtmp/')
rtmpStream.publish('35738542601194?s_asc=1&s_bl=1&s_oil=2&s_psm=1&s_pub=1&s_sw=0&s_tids=1&s_vt=api-s&a=AbwDopRUK7vfeJSz')
```

**Reply by yjjandyxy** [reply]
I used this func :
     if isAudio  {
                rtmpStream.appendSampleBuffer(sampleBuffer, withType: .audio)
                
            }else{
                rtmpStream.appendSampleBuffer(sampleBuffer, withType: .video)

            }

The  sampleBuffer is 1280 x 720  but Facebook show 360p , I need some setting for code ?

**Reply by shogo4405** [reply]
Please look at rtmpStream.videoSettings.
```
rtmpStream.videoSettings = [
    .width: 640, // video output width
    .height: 360, // video output height
    .bitrate: 160 * 1000, // video output bitrate
    .profileLevel: kVTProfileLevel_H264_Baseline_3_1, // H264 Profile require "import VideoToolbox"
    .maxKeyFrameIntervalDuration: 2, // key frame / sec
]
```
